File: JavaScript/Base7/Base7.js
/**
 * @param {number} num
 * @return {string}
 */
 // (112ms)
var convertToBase7 = function(num) {
	if(num == 0) {
		return '0';
	}
    var sign = (num < 0);
	var res = '';
	num = Math.abs(num);
	while(num != 0) {
		res = num % 7 + res;
		num = Math.floor(num / 7);
	}
	return sign ? '-' + res : res;
};

File: Python/Base7/Base7.py
class Solution(object):
    # (45ms)
    def convertToBase7(self, num):
        """
        :type num: int
        :rtype: str
        """
        if not num:
            return '0'
        sign = (num < 0)
        num = abs(num)
        res = ''
        while num != 0:
            num,  mod = divmod(num, 7)
            res += repr(mod)
        if sign:
            res += '-'
        return res[::-1]

print Solution().convertToBase7(100)
print Solution().convertToBase7(-7)
